pylabrobot/tilting/tilter.py

```diff
@@ -1,89 +1,114 @@
 import math
-from typing import Optional, cast
+from typing import List, Optional
 
-from pylabrobot.machines.machine import Machine
-from pylabrobot.resources import Coordinate, Plate, Resource
+from pylabrobot.machines import Machine
+from pylabrobot.resources import Coordinate, Plate
 
 from .tilter_backend import TilterBackend
 
 
 class Tilter(Machine):
-  """ A tilt module """
+  """ Resources that tilt plates. """
 
   def __init__(
     self,
     name: str,
-    size_x: float,
-    size_y: float,
-    size_z: float,
+    size_x: int,
+    size_y: int,
+    size_z: int,
     backend: TilterBackend,
-    category: str = "tilter",
-    model: Optional[str] = None
+    hinge_coordinate: Coordinate,
+    child_resource_location: Coordinate,
+    category: Optional[str] = None,
+    model: Optional[str] = None,
   ):
-    super().__init__(name=name, size_x=size_x, size_y=size_y, size_z=size_z, backend=backend,
-                     category=category, model=model)
+    super().__init__(
+      name=name,
+      size_x=size_x, size_y=size_y, size_z=size_z, backend=backend,
+      category=category, model=model)
     self.backend: TilterBackend = backend  # fix type
-    self._angle: int = 0
+    self._absolute_angle: float = 0
+    self._hinge_coordinate = hinge_coordinate
+    self.child_resource_location = child_resource_location
 
-  def get_plate(self) -> Plate:
-    """ Get the plate that is currently attached to the tilt module. If no plate is assigned, raise
-    a RuntimeError. """
-
-    if len(self.children) != 1:
-      raise RuntimeError("No plate on this tilt module.")
-
-    return cast(Plate, self.children[0])
+  @property
+  def absolute_angle(self) -> float:
+    return self._absolute_angle
 
-  def assign_child_resource(self, resource: Resource, location: Coordinate, reassign: bool = True):
-    if len(self.children) > 0:
-      raise RuntimeError("Tilt module already has a plate.")
-    if not isinstance(resource, Plate):
-      raise RuntimeError("Tilt module can only have plates.")
-    return super().assign_child_resource(resource=resource, location=location, reassign=reassign)
+  async def set_angle(self, absolute_angle: int):
+    """ Set the tilt module to rotate to a given angle.
 
-  @property
-  def angle(self) -> int:
-    return self._angle
+    Args:
+      absolute_angle: The absolute (unsigned) angle to set rotation to, in degrees, measured from
+        horizontal as zero.
+    """
+    # if the hinge is on the left side of the tilter, the angle is kept positive
+    # else, the angle is converted to negative. this follows Euler angle conventions.
 
-  async def set_angle(self, angle: int):
-    """ Set the tilt module to rotate by a given angle.
+    angle = absolute_angle if self._hinge_coordinate.x < self._size_x / 2 else -absolute_angle
+    await self.backend.set_angle(angle=abs(angle))
+    self._absolute_angle = absolute_angle
```

Comment:
the `angle` is not used here, it's always `absolute_angle` because `abs(angle) = absolute_angle`. Should the parameter to backend just be `angle`? I don't think it should, so we can simplify this code, but maybe you intended it differently.

Reply:
True, true, I removed the line that signs `angle` and replaced the `set_angle()` argument to `absolute_angle`.